### DISCUSSION TITLE
We need a negative feedback button 👎

### DISCUSSION THREAD

**Discussion by Undead34** [codewars/codewars.com#3058] [Ideas]
The page is pretty good, but it needs a button to give negative to the solutions and I know that this topic can be delicate, but these are necessary things.

The problem:
We have the "best practices" button, it's a system where the highest rated answer, well in theory it should be better.

Well, we know that every developer has a different solution and that's good, the problem here is that not all of them are the best and/or efficient, that's why the "best practices" button was invented.

Ok let's imagine that we have two answers, I am going to base on the first exercise "Unique In Order".

Imagine that these solutions are given:

```javascript
var uniqueInOrder=function(iterable){
    return [...iterable].filter((a, i) => a !== iterable[i-1])
}
```

This is an excellent solution to the exercise, it is copying the iterable and then creating a new array to avoid having two of the same item in a row.

But then we have this:

```javascript
var uniqueInOrder = function (iterable)
{
	return [].filter.call(iterable, (function (a, i) { return iterable[i - 1] !== a }));
}
```

This code is not the best, it adds unnecessary complexity by calling ``call``, that way of using ``function`` is not that nice. It is valid, yes, but it is not the best.

So let's suppose that 20 programmers thought of the second way, as there is only one way to vote positive, they will vote positive for the answer that looks like theirs, it is logical. 

But there may be 200 programmers who do not vote that answer because it does not seem good to them and leave it without voting, well, over time that answer will go up and up until it appears in the top of the best answers when it is not.

Don't come to me with things like:

"ah, but voting negative can make people feel bad."

No, they are supposed to be on this page to learn from their mistakes and understand why it is wrong, and they can correct it.

The solution:

Add a button to give negative, the button does not need to have a counter on it, to avoid what happened before on YouTube.

It can be like the one on Stack Overflow that accepts negative numbers to bury those bad answers and thus know that one is worse than another.

Well, you can leave your opinion here below. 😗

**Comment by Undead34** [comment]
@monadius @Just4FunCoder? explain why they think it is a bad idea.

**Reply by Kacarott** [reply]
Ironically, they have done exactly what you want to be possible in Codewars. Providing negative feedback, without any explanation or constructive criticism :)

It is already possible to provide constructive criticism on Codewars by simply commenting on solutions, explaining why they are suboptimal. This helps people who are there to learn far more than simply burying solutions. Also, you can already vote on solutions which you *do* think is good, to raise it up above the bad solutions.

**Comment by ejini6969** [comment]
@Undead34 [This issue](https://github.com/codewars/codewars.com/issues/1063) has been raised few years ago. And clearly the cons outweigh the pros as you read along the thread. Also, FYI, JS's spread operator and arrow functions are only introduced in ES6, so your reasoning in downvoting such solutions is not entirely true.

**Reply by hobovsky** [reply]
Oh, so this discussion is a duplicate. In such case, please continue on the [original ticket](https://github.com/codewars/codewars.com/issues/1063), and I will close this discussion as a duplicate.

**Comment by hobovsky** [comment]
I think that the downvote button does not solve the problem of bad, upvoted solutions. The "bad" upvotes are there because users who upvote "bad" solutions simply fail to see what is bad about them: they don;t know better syntax, or they don't know that the solution has terrible performance characteristics, or they didn't get past the stage of "one-liners are awesome" yet. Similarly, the "downvote" button would be affected by the same problem: "good" solutions would be downvoted by users who aren't able to tell why a solution is good. This way, many good solutions with good performance or which use proper built-ins instead reinventing the wheel, will end up downvoted as ["too mysterious"](https://www.codewars.com/kata/reviews/541c8b5e7e4b4c61e2000149/groups/541df7aa259d9c7b80000a88#6502d2e030881a1ae3a2d682), or as [not in spirit of the game](https://www.codewars.com/kata/541a354c39c5efa5fa001372/solutions/python#622fc17acdd705000e30f227).
